Lib/unittest/case.py

```diff
@@ -1626,3 +1629,46 @@ def shortDescription(self):
 
     def __str__(self):
         return "{} {}".format(self.test_case, self._subDescription())
+
+
+class _SubTestCleanupHelper():
+    """
+    Helper class to manage cleanups and context managers inside subTest blocks,
+    without exposing full TestCase functionality.
+    """
+
+    def __init__(self, subtest, outcome=None):
+        self._cleanups = []
+        self._subtest = subtest
+        self._outcome = outcome
+
+    def _callCleanup(self, function, /, *args, **kwargs):
+        function(*args, **kwargs)
```

Comment:
```suggestion
    @staticmethod
    def _callCleanup(function, /, *args, **kwargs):
        function(*args, **kwargs)
```

I don't think we need to hold a reference to `self` so a staticmethod can be used but OTOH, I don't know if there is a world where we want to be able to subclass this one in the future.